logutils/pre_login_log.go

```diff
@@ -0,0 +1,53 @@
+package logutils
+
+import (
+	"errors"
+	"path/filepath"
+)
+
+const(
+	defaultPreLoginLogFile  = "pre_login.log"
+	defaultPreLoginLogLevel = "ERROR"
+	defaultPreLoginLogEnabled = true
+)
+
+type PreLoginLog struct {
+	enabled bool
+	level   string
+	// absolute path to the log directory, it should be the same as node config's logDir
+	logDir string
+}
+
+func NewPreLoginLog() *PreLoginLog {
+	return &PreLoginLog{
+		enabled: defaultPreLoginLogEnabled,
+		level:   defaultPreLoginLogLevel,
+	}
+}
+
+func (l *PreLoginLog) SetEnabled(enabled bool) {
+	l.enabled = enabled
+}
+
+func (l *PreLoginLog) SetLevel(level string) error {
+	switch level {
+	case "ERROR", "WARN", "INFO", "DEBUG", "TRACE":
```

Comment:
There are some duplicates in status-go referring to the enumeration of supported log levels. I'm thinking if we should eventually use a consistent place in the code as reference. The example below isn't the place to use as a central reference, but it shows we already have a const somewhere enumerating log levels and that it would be good to refer constants or even to a level type instead of raw strings:

https://github.com/status-im/status-go/blob/c5fdd30781634ce88265531ea9dd7682c770d5bf/protocol/requests/set_log_level.go#L7-L12